From e9ccda5a1cb3ad9020700f9de8867c1c22c9d34e Mon Sep 17 00:00:00 2001
From: Shawn Xiao <b49994@freescale.com>
Date: Wed, 26 Aug 2015 16:02:00 +0800
Subject: [PATCH] MGS-955-3 [#1938]remove IRQF_DISABLED

The IRQF_DISABLED has been removed from 4.1.0 kernel. To accomodate with
the change, add version check logic and use 0x0 instead of IRQF_DISABLED
from 4.1.0 kernel on.

Date Sep 14, 2015

Signed-off-by: Shawn Xiao <b49994@freescale.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_device.c     | 38 +++++++++++++++++++---
 1 file changed, 33 insertions(+), 5 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
index 37462fa..aae964e 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_device.c
@@ -2114,8 +2114,12 @@ gckGALDEVICE_Setup_ISR(
 #else
 #if gcdMULTI_GPU
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLine3D[gcvCORE_3D_0_ID], isrRoutine3D0, IRQF_DISABLED,
-        "galcore_3d_0", Device
+#else
+        Device->irqLine3D[gcvCORE_3D_0_ID], isrRoutine3D0, 0x0,
+#endif
+     	"galcore_3d_0", Device
         );
 
     if (ret != 0)
@@ -2135,7 +2139,11 @@ gckGALDEVICE_Setup_ISR(
 
 #if gcdMULTI_GPU > 1
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLine3D[gcvCORE_3D_1_ID], isrRoutine3D1, IRQF_DISABLED,
+#else
+        Device->irqLine3D[gcvCORE_3D_1_ID], isrRoutine3D1, 0x0,
+#endif
         "galcore_3d_1", Device
         );
 
@@ -2156,8 +2164,12 @@ gckGALDEVICE_Setup_ISR(
 #endif
 #elif gcdMULTI_GPU_AFFINITY
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLines[gcvCORE_MAJOR], isrRoutine3D0, IRQF_DISABLED,
-        "galcore_3d_0", Device
+#else
+        Device->irqLines[gcvCORE_MAJOR], isrRoutine3D0, 0x0,
+#endif
+	"galcore_3d_0", Device
         );
 
     if (ret != 0)
@@ -2176,7 +2188,11 @@ gckGALDEVICE_Setup_ISR(
     Device->isrInitializeds[gcvCORE_MAJOR] = gcvTRUE;
 
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLines[gcvCORE_OCL], isrRoutine3D1, IRQF_DISABLED,
+#else
+	Device->irqLines[gcvCORE_OCL], isrRoutine3D1, 0x0,
+#endif
         "galcore_3d_1", Device
         );
 
@@ -2196,8 +2212,12 @@ gckGALDEVICE_Setup_ISR(
     Device->isrInitializeds[gcvCORE_OCL] = gcvTRUE;
 #else
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLines[gcvCORE_MAJOR], isrRoutine, IRQF_DISABLED,
-        "galcore interrupt service", Device
+#else
+	Device->irqLines[gcvCORE_MAJOR], isrRoutine, 0x0,
+#endif
+    	"galcore interrupt service", Device
         );
 
     if (ret != 0)
@@ -2255,8 +2275,12 @@ gckGALDEVICE_Setup_ISR_2D(
         );
 #else
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLines[gcvCORE_2D], isrRoutine2D, IRQF_DISABLED,
-        "galcore interrupt service for 2D", Device
+#else
+        Device->irqLines[gcvCORE_2D], isrRoutine2D, 0x0,
+#endif
+	"galcore interrupt service for 2D", Device
         );
 #endif
 
@@ -2313,8 +2337,12 @@ gckGALDEVICE_Setup_ISR_VG(
         );
 #else
     ret = request_irq(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         Device->irqLines[gcvCORE_VG], isrRoutineVG, IRQF_DISABLED,
-        "galcore interrupt service for 2D", Device
+#else
+        Device->irqLines[gcvCORE_VG], isrRoutineVG, 0x0,
+#endif
+	"galcore interrupt service for 2D", Device
         );
 #endif
 
-- 
1.8.0

